WalletWasabi/Crypto/GroupElement.cs

```diff
@@ -59,5 +60,24 @@ public GroupElement(GEJ groupElement)
 		}
 
 		public static bool operator !=(GroupElement a, GroupElement b) => !(a == b);
+
+		public Scalar Sha256(GroupElement groupElement)
+		{
+			Guard.NotNull(nameof(groupElement), groupElement);
+			Guard.False($"{nameof(groupElement)}.{nameof(groupElement.IsInfinity)}", groupElement.IsInfinity);
+
+			Guard.False($"{nameof(IsInfinity)}", IsInfinity);
+
+			var concatenation = ByteHelpers.Combine(
+				Ge.x.ToBytes(),
+				Ge.y.ToBytes(),
+				groupElement.Ge.x.ToBytes(),
+				groupElement.Ge.y.ToBytes());
+
+			using var sha256 = System.Security.Cryptography.SHA256.Create();
+			var hash = sha256.ComputeHash(concatenation);
+
+			return new Scalar(hash);
```

Comment:
Here is can overflow, yes. But take into account that we want to `challenge` to be a Scalar and it doesn't matter if it overflows or not because it will be multiplied by "something" so, overflowing is not an issue.

On the other hand you "almost" never want the hash of a single group element but the hash of a "concatenation" (we can define it as a binary concatenation, addition or some other way of composition) of all the public points involved in the proof, most of the times they are many.

About concatenation: everywhere a symbol like `|` is used to indicate that a "concatenation" is required but it doesn't have to be like that, i mean, imagine you are creating a proof of "something" and that involves `G1`, `G2` and `G3`, you could do `challenge = Hash( TextToG("proof of something") * (G1 + G2 + G3) )` or something like that. This is not a recomendation, this is just informative.